src/main/java/bio/terra/service/iam/IamService.java

```diff
@@ -27,11 +33,20 @@
 @Component
 public class IamService {
     private final IamProviderInterface iamProvider;
+    private final ConfigurationService configurationService;
 
     @Autowired
-    public IamService(IamProviderInterface iamProvider) {
+    public IamService(IamProviderInterface iamProvider,
+                      ConfigurationService configurationService) {
         this.iamProvider = iamProvider;
+        this.configurationService = configurationService;
     }
+    int cacheSize = 100;
+    // TODO is this getting called too early? configurationService.getParameterValue(AUTH_CACHE_SIZE);
```

Comment:
can i set this as a config this early?

Reply:
It probably makes sense to create the `authorizedMap` in the constructor.